### PR TITLE
feat: validate invoices fetched for offers

### PR DESCRIPTION
**Pull request by michael1011**

### CODE REVIEW

File: src/utils/invoice.ts [hunk 1]
```diff
@@ -248,3 +248,48 @@ export const isBolt12Offer = (offer: string) => {
...
+
+        for (const signer of expectedSigners) {
+            if (signer.length !== invoiceSigner.length) {
+                continue;
+            }
+
+            if (signer.every((b, i) => b === invoiceSigner[i])) {
+                return true;
+            }
+            console.log("no match");
```
Comment (maybeast) [inline src/utils/invoice.ts:288]: remove or add more context. In the current state it's pretty useless.

File: src/utils/invoice.ts [hunk 2]
```diff
@@ -248,3 +248,48 @@ export const isBolt12Offer = (offer: string) => {
...
+        expectedSigners.push(of.signing_pubkey);
+    }
+
+    for (const path of of.paths) {
+        const hops = path.hops;
+        if (hops.length > 0) {
+            expectedSigners.push(hops[hops.length - 1].pubkey);
+        }
+
+        for (const hop of hops) {
```
Comment (maybeast) [inline src/utils/invoice.ts:266]:
```suggestion
        hops.forEach(hop => hop.free());
```

**Review by maybeast — APPROVED** [review]

**Comment by kilrau** [comment]
2/10 of my testnet bip353 resolve tests needed too long to wake up my phoenix app. Happened when I restarted my phone to make sure Phoenix is not running in the background already. But the notification literally came in a second or two after web app timed out with "failed to fetch". I suggest to bump the timeout for about 3s.

**Review by kilrau — CHANGES_REQUESTED** [review]
Refund qr code generation fails (json works). Also we should throw a tool tip error if that happens and block the jump to the next screen to not let users create swaps without refund file.

image

**Review by kilrau — APPROVED** [review]
Moved above comments into separate issues. Invoice <> Offer validation works.

**Pull request merged**